blocks/ad-taboola-block/features/ad-taboola/default.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import getProperties from 'fusion:properties';
+import Consumer from 'fusion:consumer';
+
+const taboolaLoader = (publisherID) => (
+  `window._taboola = window._taboola || [];
+    _taboola.push({article:'auto'});
+    !function (e, f, u, i) {
+      if (!document.getElementById(i)){
+        e.async = 1;
+        e.src = u;
+        e.id = i;
+        f.parentNode.insertBefore(e, f);
+      }
+    }(document.createElement('script'),
+    document.getElementsByTagName('script')[0],
+    '//cdn.taboola.com/libtrc/${publisherID}/loader.js',
+    'tb_loader_script');
+    if(window.performance && typeof window.performance.mark == 'function')
+      {window.performance.mark('tbl_ic');}
+  `
+);
+
+const taboolaFlusher = () => (
+  `
+  window._taboola = window._taboola || [];
+  _taboola.push({flush: true});
+  `
+);
+
+@Consumer
+class AdTaboola extends Component {
+  constructor(props) {
+    super(props);
+
+    const { arcSite } = this.props;
+    this.publisherID = getProperties(arcSite).taboolaPublisherId;
+    const { customFields: { placement, mode, container } = {} } = props;
+
+    this.state = {
+      placement,
+      mode,
+      container,
+    };
+  }
+
+  componentDidMount() {
+    if (!this.canRender()) {
+      return;
+    }
+
+    this.insertLoader();
+    this.insertFlusher();
+  }
+
+  appendScript = (name, container, sourceResolver) => {
+    const script = document.createElement('script');
+    script.id = name;
+    script.async = true;
+    script.type = 'text/javascript';
+    script.innerHTML = sourceResolver();
+
+    container.appendChild(script);
+  }
+
+  canRender() {
+    const { placement, mode, container } = this.state;
+    return (this.publisherID && placement && mode && container);
+  }
+
+  insertLoader() {
+    const loader = document.getElementById('tbl-loader');
+    if (loader) {
+      return;
+    }
+    const head = document.getElementsByTagName('head')[0];
+    if (!head) {
+      return;
+    }
+
+    this.appendScript('tbl-loader', head, () => taboolaLoader(this.publisherID));
+  }
+
+  insertFlusher() {
+    const flusher = document.getElementById('tbl-flusher');
+    if (flusher) {
+      return;
+    }
+    const body = document.getElementsByTagName('body')[0];
+    if (!body) {
+      return;
+    }
+
+    this.appendScript('tbl-flusher', body, taboolaFlusher);
```

Comment:
Just wondering if this might need to check for window undefined before accessing the DOM?  In other words, when this initially runs SSR...is this ok?

Reply:
The `appendScript` function is called throw `insertLoader` and `insertFlusher`, and both are called on `componentDidMount` that is only execute client side.

SSR just will render the main container inside the page, but only if all the parameters are setup (publisherID and block parameters)